### PR TITLE
Simplify frontend build

### PR DESCRIPTION
**Pull request by gandreadis**
This PR refactors parts of the Docker build procedure, for a simpler and cleaner process.

Specifically, it offers the following changes:

* All configuration-specific details are moved to a JSON file. The `README` now provides a template for this `keys.json` file, which then needs to be present in the `opendc` root folder at build-time. The main difference in the Docker build process is that the `docker run` command no longer needs the OAuth client ID and server URL as inputs, since these are read from that configuration file.
* The OAuth client secret has been permanently removed from all Docker configuration steps, as it is never used by the web-server.
* The newly linked frontend-module now supports custom URL-definition. This way, the previously hard-coded server URL can be replaced with another one (such as `http://localhost:8081`), for customized setups. This base URL can be set in the `keys.json` configuration file.
* The manual Docker-string-replacements in `configure.sh` have been rendered obsolete and removed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,20 +46,36 @@ Open VirtualBox, navigate to the settings of your default docker VM, and go to t
...
 
 ```bash
 # Clone the repo and its submodules
 git clone --recursive https://github.com/atlarge-research/opendc.git
 
 # Enter the directory
 cd opendc/
+```
+
+In the directory you just entered, you need to set up a small configuration file. To do this, create a file called `keys.json` in the `opendc` folder. In this file, simply replace `your-google-oauth-client-id` with your `client_id` from the OAuth client ID you created.
```
Comment (leonoverweel) [inline README.md]: Can you also clarify what exactly needs to be in the `ROOT_DIR` field, here?

File: README.md [hunk 2]
```diff
@@ -46,20 +46,36 @@ Open VirtualBox, navigate to the settings of your default docker VM, and go to t
...
+
+```json
+{
+  "OAUTH_CLIENT_ID": "your-google-oauth-client-id",
+  "SERVER_BASE_URL": "http://localhost:8081",
+  "ROOT_DIR": "/opendc",
+  "FLASK_SECRET": "This is a super duper secret flask key",
+  "DATABASE_LOCATION": "/data/database/opendc.db"
+}
```
Comment (leonoverweel) [inline README.md]: There should be alphabetized.

File: README.md [hunk 3]
```diff
@@ -46,20 +46,36 @@ Open VirtualBox, navigate to the settings of your default docker VM, and go to t
...
+```
+
+Once this configuration file is set up, we're ready to get the install up and running:
```
Comment (leonoverweel) [inline README.md]: To conform with the writing style in the rest of the `README`, suggested rephrase to: "Now, start the server:"

(I don't think the first dependent clause is necessary.)

**Comment by gandreadis** [comment]
Thanks for the feedback @leonoverweel! I've implemented it, please have a look if my changes address your points.

**Comment by sacheendra** [comment]
@gandreadis The changes look good and everything works fine. 

But, I think we might also need to support a system using environment variables in the future so that it can be deployed directly from dockerhub instead of requiring a clone (Using a keys.json files requires that). That something for the future though.

**Pull request merged**